frontend/src/js/external-forms/FormConfigLoader.tsx

```diff
@@ -0,0 +1,181 @@
+import styled from "@emotion/styled";
+import type { StateT } from "app-types";
+import { FC, useState, useEffect, memo, ReactNode } from "react";
+import { useFormContext } from "react-hook-form";
+import { useTranslation } from "react-i18next";
+import { useSelector, useDispatch } from "react-redux";
+
+import { useGetFormConfig } from "../api/api";
+import type { SelectOptionT } from "../api/types";
+import { DNDType } from "../common/constants/dndTypes";
+import { useDatasetId } from "../dataset/selectors";
+import { Language, useActiveLang } from "../localization/useActiveLang";
+import type { FormConfigT } from "../previous-queries/list/reducer";
+import { setMessage } from "../snack-message/actions";
+import Dropzone from "../ui-components/Dropzone";
+
+import { setExternalForm } from "./actions";
+import type { Form, FormField } from "./config-types";
+import type { FormConceptGroupT } from "./form-concept-group/formConceptGroupState";
+import { collectAllFormFields } from "./helper";
+import { selectActiveFormType, selectFormConfig } from "./stateSelectors";
+import type { DragItemFormConfig } from "./types";
+
+const Root = styled("div")`
+  display: flex;
+  align-items: center;
+  border-radius: ${({ theme }) => theme.borderRadius};
+`;
+
+const SxDropzone = styled(Dropzone)`
+  padding: 10px 20px 20px 10px;
+  color: ${({ theme }) => theme.col.black};
+`;
+
+const DROP_TYPES = [DNDType.FORM_CONFIG];
+
+// Potentially transform the stored field value to support older saved form configs
+//
+// because we changed the SELECT values:
+// from string, e.g. 'next'
+// to SelectValueT, e.g. { value: 'next', label: 'Next' }
+//
+// and because we introduced the DNDTypes (CONCEPT_LIST)
+const transformLoadedFieldValue = (
+  field: FormField,
+  value: unknown,
+  {
+    activeLang,
+    datasetOptions,
+  }: { activeLang: Language; datasetOptions: SelectOptionT[] },
+) => {
+  switch (field.type) {
+    case "CONCEPT_LIST":
+      return (value as FormConceptGroupT[]).map((group) => ({
+        ...group,
+        concepts: group.concepts.map((concept) => ({
+          ...concept,
+          type: DNDType.CONCEPT_TREE_NODE,
+        })),
+      }));
+    case "DATASET_SELECT":
+      if (typeof value === "object") return value as SelectOptionT;
+      if (typeof value === "string") {
+        return datasetOptions.find((option) => option.value === value);
+      }
+
+      return value;
+    case "SELECT":
+      if (typeof value === "object") return value as SelectOptionT;
+      if (typeof value === "string") {
+        const options = field.options.map((option) => ({
+          label: option.label[activeLang] || "",
```

Comment:
Wouldn't it make more sense to use any fallback value (eg. the first value in label: `values(option.labels).shift()`) instead of displaying nothing?

Reply:
Caveat being that I don't know if you generally fallback onto value and if it'd be preferable to accidentally display spanish instead of a potentially weird technical term.